contract_manager/src/contracts/near.ts

```diff
@@ -0,0 +1,264 @@
+      methodName: "get_price_unsafe",
+      args: { price_identifier: feedId },
+    });
+    const emaPrice: {
+      price: string;
+      conf: string;
+      expo: number;
+      publish_time: number;
+    } | null = await account.viewFunction({
+      contractId: this.address,
+      methodName: "get_ema_price_unsafe",
+      args: { price_id: feedId },
+    });
+    if (price === null || emaPrice === null) {
+      return undefined;
+    } else {
+      return {
+        price: {
+          price: price.price,
+          conf: price.conf,
+          expo: price.expo.toString(),
+          publishTime: price.publish_time.toString(),
+        },
+        emaPrice: {
+          price: emaPrice.price,
+          conf: emaPrice.conf,
+          expo: emaPrice.expo.toString(),
+          publishTime: emaPrice.publish_time.toString(),
+        },
+      };
+    }
+  }
+
+  async executeUpdatePriceFeed(
+    senderPrivateKey: PrivateKey,
+    vaas: Buffer[]
+  ): Promise<TxResult> {
+    if (vaas.length === 0) {
+      throw new Error("no vaas specified");
+    }
+    const senderAddress = await this.chain.getAccountAddress(senderPrivateKey);
+    const account = await this.chain.getNearAccount(
+      senderAddress,
+      senderPrivateKey
+    );
+    const results = [];
+    for (const vaa of vaas) {
+      const outcome = await account.functionCall({
+        contractId: this.address,
+        methodName: "update_price_feeds",
+        args: { data: vaa.toString("hex") },
+        gas: new BN(300e12),
+        attachedDeposit: new BN(1e12),
+      });
+      results.push({ id: outcome.transaction.hash, info: outcome });
+    }
+    if (results.length === 1) {
+      return results[0];
+    } else {
+      return {
+        id: results.map((x) => x.id).join(","),
+        info: results.map((x) => x.info),
+      };
+    }
+  }
+
+  async executeGovernanceInstruction(
+    senderPrivateKey: PrivateKey,
+    vaa: Buffer
+  ): Promise<TxResult> {
+    const senderAddress = await this.chain.getAccountAddress(senderPrivateKey);
+    const account = await this.chain.getNearAccount(
+      senderAddress,
+      senderPrivateKey
+    );
+    const outcome = await account.functionCall({
+      contractId: this.address,
+      methodName: "execute_governance_instruction",
+      args: { vaa: vaa.toString("hex") },
+      gas: new BN(300e12),
+      attachedDeposit: new BN(1e12),
+    });
+    return { id: outcome.transaction.hash, info: outcome };
+  }
+
+  getBaseUpdateFee(): Promise<{ amount: string; denom?: string }> {
+    throw new Error("near contract doesn't implement getBaseUpdateFee method");
+  }
+  getLastExecutedGovernanceSequence(): Promise<number> {
+    throw new Error(
+      "near contract doesn't implement getLastExecutedGovernanceSequence method"
+    );
+  }
+  getGovernanceDataSource(): Promise<DataSource> {
+    throw new Error(
+      "near contract doesn't implement getGovernanceDataSource method"
+    );
+  }
```

Comment:
do you know how the upgrade is going to work? we initially execute governance instruction, but what can we do after it :?

Reply:
Looks like we'll need to call a permissionless method `update_contract` with the new wasm file as input.